v-next/hardhat-viem/README.md

```diff
@@ -0,0 +1,269 @@
+[![npm](https://img.shields.io/npm/v/@nomicfoundation/hardhat-viem.svg)](https://www.npmjs.com/package/@nomicfoundation/hardhat-viem) [![hardhat](https://hardhat.org/buidler-plugin-badge.svg?1)](https://hardhat.org)
+
+# hardhat-viem
+
+[Hardhat](https://hardhat.org) plugin for integration with [Viem](https://github.com/wagmi-dev/viem), a lightweight, composable, and type-safe Ethereum library.
+
+## What
+
+This plugin integrates the Viem Ethereum library into your Hardhat development environment. Viem is an alternative to [ethers.js](https://docs.ethers.io/) that offers developers a different way to interact with the Ethereum blockchain.
+
+By installing and configuring `hardhat-viem`, you gain access to the capabilities of the Viem library directly within your Hardhat projects. This integration enables you to perform various Ethereum-related tasks using Viem's features and functionalities.
+
+Note: This plugin relies on the Viem library, so familiarity with [Viem's documentation](https://viem.sh/docs/getting-started.html) can enhance your experience when working with `hardhat-viem`.
+
+## Installation
+
+```bash
+npm install --save-dev @nomicfoundation/hardhat-viem viem
+```
+
+And add the following statement to your `hardhat.config.js`:
+
+```js
+require("@nomicfoundation/hardhat-viem");
+```
+
+Or, if you are using TypeScript, add this to your `hardhat.config.ts`:
+
+```js
+import "@nomicfoundation/hardhat-viem";
+```
+
+**Note:** you might want to pin Viem-related dependencies because Viem does not strictly follow semantic versioning for type changes. You can read more [here](https://hardhat.org/hardhat-runner/docs/advanced/using-viem#managing-types-and-version-stability).
+
+## Required plugins
+
+No plugins dependencies.
+
+## Tasks
+
+This plugin creates no additional tasks.
+
+## Environment extensions
+
+This plugins adds a `viem` object to the Hardhat Runtime Environment which provides a minimal set of capabilities for interacting with the blockchain.
```

Comment:
This section needs to be reworked.

I would do it minimally - we are going to review all these readmes. But we shouldn't be factually misleading.

Reply:
Done [here](https://github.com/NomicFoundation/hardhat/pull/5825/commits/1c81b6cbd79a33d3bde26da3554911710bad5884).